addons/mail/controllers/user.py

```diff
@@ -0,0 +1,18 @@
+
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from werkzeug.exceptions import NotFound
+
+from odoo import http
+from odoo.http import request
+
+
+class UserController(http.Controller):
+    @http.route("/mail/user/get_user_info", methods=["POST"], type="json", auth="user")
+    def mail_get_user_info(self, user_id, fields=None, **kwargs):
+        """Get user information by user_id."""
```

Comment:
I would not name the controller in a generic way, because this feels over-engineered to the problem at hand.
Personally I would prefer focus on the problem at hand the most straight-forward way possible: we want to retrieve user info from avatar card, let's just name it with this specificity in mind, therefore something like `mail/avatar_card/get_user_infos`. No need to provide the fields as param, the route can have it in clear.

This may make sense in future to have generic route, but I would prefer such consideration when this makes sense, not rushing prematurely on an over-engineered solution when a much simpler solution is more than enough.